pycbc/population/rates_functions.py

```diff
@@ -0,0 +1,443 @@
+    min_mass = kwargs.get('min_mass', 5.)
+    max_mass = kwargs.get('max_mass', 95.)
+    alpha = kwargs.get('alpha', -2.35)
+    max_mtotal = min_mass + max_mass    
+    m1, m2 = np.array(m1), np.array(m2)    
+    
+    C_imf = max_mass**(alpha + 1)/(alpha + 1) - min_mass**(alpha + 1)/(alpha + 1)    
+    
+    xx = np.minimum(m1, m2)
+    m1 = np.maximum(m1, m2)
+    m2 = xx    
+    
+    bound = np.sign(max_mtotal - m1 - m2) + np.sign(max_mass - m1) * np.sign(m2 - min_mass)
+    idx = np.where(bound != 2)
+    
+    p_m1_m2 = np.zeros_like(m1)
+    p_m1_m2[m1 <= max_mtotal/2.] = (1./C_imf) * m1[m1 <= max_mtotal/2.]**alpha /(m1[m1 <= max_mtotal/2.] - min_mass)
+    p_m1_m2[m1 > max_mtotal/2.] = (1./C_imf) * m1[m1 > max_mtotal/2.]**alpha /(max_mass - m1[m1 > max_mtotal/2.])
+    p_m1_m2[idx] = 0    
+    
+    return p_m1_m2/2.
+
+# Functions to generate samples for the two canonical models
+def draw_imf_samples(**kwargs):
+    ''' Draw samples for power-law model
+
+        Parameters
+        ----------
+        **kwargs: string
+           Keyword arguments as model parameters and number of samples
+
+        Returns
+        -------
+        array
+           The first mass
+        array
+           The second mass
+    '''
+ 
+    alpha_salpeter = kwargs.get('alpha', -2.35)
+    nsamples = kwargs.get('nsamples', 1)
+    min_mass = kwargs.get('min_mass', 5.)
+    max_mass = kwargs.get('max_mass', 95.)
+    max_mtotal = min_mass + max_mass
+        
+    a = (max_mass/min_mass)**(alpha_salpeter + 1.0) - 1.0
+    beta = 1.0 / (alpha_salpeter + 1.0)
+    
+    k = nsamples * int(1.5 + np.log(1 + 100./nsamples))
+    aa = min_mass * (1.0 + a * np.random.random(k))**beta
+    bb = np.random.uniform(min_mass, aa, k) 
+    
+    idx = np.where(aa + bb < max_mtotal)
+    m1, m2 = (np.maximum(aa, bb))[idx], (np.minimum(aa, bb))[idx]
+    
+    return np.resize(m1, nsamples), np.resize(m2, nsamples)    
+
+def draw_lnm_samples(**kwargs):
+    ''' Draw samples for uniform-in-log model
+
+        Parameters
+        ----------
+        **kwargs: string
+           Keyword arguments as model parameters and number of samples
+
+        Returns
+        -------
+        array
+           The first mass
+        array
+           The second mass
+    '''
+
+    #PDF doesnt match with sampler
+    nsamples = kwargs.get('nsamples', 1)
+    min_mass = kwargs.get('min_mass', 5.)
+    max_mass = kwargs.get('max_mass', 95.)
+    max_mtotal = min_mass + max_mass    
+    lnmmin = np.log(min_mass)
+    lnmmax = np.log(max_mass)
+    
+    k = nsamples * int(1.5 + np.log(1 + 100./nsamples))
+    aa = np.exp(np.random.uniform(lnmmin, lnmmax, k))
+    bb = np.exp(np.random.uniform(lnmmin, lnmmax, k))
+  
+    idx = np.where(aa + bb < max_mtotal)
+    m1, m2 = (np.maximum(aa, bb))[idx], (np.minimum(aa, bb))[idx]
+    
+    return np.resize(m1, nsamples), np.resize(m2, nsamples)
+
+def m1m2_to_mcheta(m1, m2):
+    ''' Get chirp mass and eta for m1 and m2
```

Comment:
Please use double quotes here.

Reply:
This function has been replaced by one in pycbc.conversions